File: cpp/hardware/cooling/Cooling.cpp
#pragma once
#include "../../component/Component.cpp"
#include "../../component/Hardware.cpp"
#include "../../interfaces/Bootable.cpp"

class Cooling : public Hardware, public Component, public Bootable
{
protected:
    int maxThermalDissipation;

public:
    // Constructor
    Cooling() : Hardware(), Component(), maxThermalDissipation(0) {}
    Cooling(int thermalDissipation) : Hardware(), Component(), maxThermalDissipation(thermalDissipation) {}
    Cooling(const std::string &serialNumber, const Date &installationDate, bool physicallyInstalled, int thermalDissipation, const std::string &id, const std::string &manufacturer, const std::string &model, double price)
        : Hardware(serialNumber, installationDate, physicallyInstalled), Component(id, manufacturer, model, price), maxThermalDissipation(thermalDissipation) {}

    // Destructor
    virtual ~Cooling();

    // Setters
    void setMaxThermalDissipation(int maxThermalDissipation);

    // Getters
    int getMaxThermalDissipation() const;
};

File: python/hardware/cooling/cooling.py
from datetime import date
from components.component import Component
from components.hardware import Hardware

class Cooling(Hardware, Component):
    def __init__(self, serial_number="", installation_date=None, physically_installed=False, max_thermal_dissipation=0, id="", manufacturer="", model="", price=0.0):
        Hardware.__init__(self, serial_number, installation_date, physically_installed)
        Component.__init__(self, id, manufacturer, model, price)
        self._max_thermal_dissipation = max_thermal_dissipation

    # Setters
    def set_max_thermal_dissipation(self, max_thermal_dissipation):
        self._max_thermal_dissipation = max_thermal_dissipation

    # Getters
    def get_max_thermal_dissipation(self):
        return self._max_thermal_dissipation